// @ELL19476/analytics-extended/index.js
class AnalyticsClient {
    constructor(config = {}) {
        this.version = '2.8.2'; // This version has the bug
        this.initialized = false;
        this.config = {
        batchSize: 10,
        flushInterval: 5000,
        ...config
        };
        this.eventQueue = [];
    }

    track(eventName, metadata = {}) {
        // Bug in 2.8.0: TypeError when metadata contains Date objects
        if (metadata.timestamp instanceof Date) {
        // The bug: trying to use a non-existent method
        metadata.timestamp.toAnalyticsFormat();
        }

        this.eventQueue.push({
        name: eventName,
        ...metadata,
        recordedAt: new Date()
        });

        return true;
    }

    pageView(pageName, properties = {}) {
        return this.track('page_viewed', {
        page: pageName,
        ...properties
        });
    }
}

let clientInstance = null;

function initializeCustomAnalytics(config = {}) {
    if (!clientInstance) {
        clientInstance = new AnalyticsClient(config);
    }
    return clientInstance;
}

module.exports = {
    initializeCustomAnalytics,
    version: '2.8.0'
};
